src/gzchunked/mod.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! Utils for forming gzchunked streams.
+
+use std::vec::Vec;
+use std::io::Write;
+use flate2::write::GzEncoder;
+
+const BLOCK_SIZE: usize = 10 << 20;
+
+pub struct GzChunked {
+    encoder: GzEncoder<Vec<u8>>,
+}
+
+impl GzChunked {
+    pub fn new() -> GzChunked {
+        GzChunked {
+            encoder: GzEncoder::new(Vec::new(), flate2::Compression::default())
```

Comment:
Prefer to use some specific compression level, rather to delegate this to whatever the library author chooses. Here, the default level is `6`, which I think is a sensible choice (but something like `5` would be better, I think), but if they decide to change it to something bigger at some point, RRG suddenly will become slower. It also hides this information from other developers and they are forced to dive into the library source to figure it out. An explicit value is better.

The timeline action should be first and foremost fast but, if possible, it should not blow up the response size too much either. I think some benchmarking is needed to check what level gives appropriate balance.